mobile/examples/phi-3/ios/LocalLLM/LocalLLM/README.md

```diff
@@ -0,0 +1,106 @@
+
+1. macOS 14+
+
+2. Xcode 15+ (latest Xcode version perferred.)
+
+3. iOS SDK 16.x + (iPhone 14 or iPhone 15 powered by a A16 or A17 preferred)
+
+**Note**: 
+  The current Xcode project contains a built .dylib for ORT and ORT GenAI. The following steps `A, B, C` under `step 1.` for building from source for the libraries are optional.
+  However if you want to build from source to include the latest updates, please use the `step 1.` as a reference.
+
+### 1. Steps to build from source for ONNX Runtime and Generative AI libraries [Optional]
+
+#### **A. Preparation**
+
+   - Install Python 3.10+
+
+   - Install flatbuffers
+     ```
+       pip3 install flatbuffers
+     ```
+
+   - Install [CMake](https://cmake.org/download/)
+
+#### **B. Compiling ONNX Runtime for iOS**
+
+```bash
+
+git clone https://github.com/microsoft/onnxruntime.git
+
+cd onnxruntime
+
+./build.sh --build_shared_lib --skip_tests --parallel --build_dir ./build_ios --ios --apple_sysroot iphoneos --osx_arch arm64 --apple_deploy_target 16.6 --cmake_generator Xcode --config Release
+
+```
+
+***Notice***
+
+  1. Before compiling, you must ensure that Xcode is configured correctly and set it on the terminal
+
+```bash
+
+sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer 
+
+```
+
+  2. ONNX Runtime needs to be compiled based on different platforms. For iOS, you can compile for arm64 or x86_64 based on needs. If you are running an iOS simulator on an Intel mac, compile for x86_64. Use arm64 for an ARM based mac to run the simulator, and to run on an iPhone.
+   
+  3. It is recommended to directly use the latest iOS SDK for compilation. Of course, you can also lower the version to be compatible with past SDKs.
+
+#### **C. Compiling Generative AI with ONNX Runtime for iOS**
+
+```bash
+
+git clone https://github.com/microsoft/onnxruntime-genai
+
+cd onnxruntime-genai
+
+python3 build.py --parallel --build_dir ./build_iphoneos --ios --ios_sysroot iphoneos --ios_arch arm64 --ios_deployment_target 16.6 --cmake_generator Xcode
+
+```
+
+#### **D. Copy over latest header files and required .dylibs built from source**
+
+If you build from source and get the latest .dylibs for ORT and ORT GenAI, please copy the .dylibs over to `mobile\examples\phi-3\ios\LocalLLM\LocalLLM\lib` and copy the latest header files over to `mobile\examples\phi-3\ios\LocalLLM\LocalLLM\header` 
+
+Usually the build output path for libonnxruntime.dylib is under `<ORT_PROJECT_ROOT>/build/intermediates/<platform>_<arch>/<build_config>/<build_config-platform>/libonnxruntime.dylib` and the build output path for libonnxruntime-genai.dylib is under `<ORT_GENAI_PROJECT_ROOT>/build/<build_config-platform>/libonnxruntime-genai.dylib`. For example: 
+it may look like: `onnxruntime/build/intermediates/iphoneos_arm64/Release/Release-iphoneos/libonnxruntime.1.19.0.dylib`
+and similarly `onnxruntime-genai/build/Release/Release-iphoneos/libonnxruntime-genai.dylib`.
```

Comment:
nit: We need to fix the genai ios build to include the architecture in the output path.

Reply:
right good catch.. - forgot how xcode generate those path, but possibly a build variable is not correctly set up for "ARCHS"